cmd/nvidia-ctk/config/config.go

```diff
@@ -72,9 +76,16 @@ func (m command) build() *cli.Command {
 		},
 		&cli.StringSliceFlag{
 			Name:        "set",
-			Usage:       "Set a config value using the pattern key=value. If value is empty, this is equivalent to specifying the same key in unset. This flag can be specified multiple times",
+			Usage:       "Set a config value using the pattern key=value. If value is empty, this is equivalent to specifying the same key in unset. This flag can be specified multiple times. If the setting represents a slice, the elements are semi-colon-separated.",
```

Comment:
```suggestion
			Usage:       "Set a config value using the pattern key=value. If value is empty, this is equivalent to specifying the same key in unset. This flag can be specified multiple times. If the setting represents a list, the elements are semi-colon-separated.",
```

Reply:
Was semicolon the default previously? I'm still trying to wrap my head around the previous behaviour vs. the new behaviour.

What I _think_ you are saying in another comment is that we previously we always interpreted every input as a single string, and if it was *supposed* to be slice, we did the separation ourselves. What's not clear (assuming I got this right) is whether we previously did the separation with semicolons or commas, and (if it was commas), why we con't continue to use commas now instead of semilcolons.

Reply:
Yes, we were previously treating a comma-separated argument as a single string (due to a bug in urfave) and we were splitting this by commas ourselves to construct a slice to insert into the config file. We can't continue to use commas now because this urfave already assumes commas as the separator for repeated args. We could consider overriding this instead, but we can only do this at an Application level and not at a Command level. It may be possible with the alpha v3.0.0 release, but I didn't want to rely on an unstable dependency.

Consider the following code:
```
package main

import (
	"fmt"
	"os"

	"github.com/urfave/cli/v2"
)

func main() {
	var slice cli.StringSlice

	c := cli.NewApp()
	c.Flags = []cli.Flag{
		&cli.StringSliceFlag{
			Name:        "slice-flag",
			Destination: &slice,
		},
	}

	c.Action = func(ctx *cli.Context) error {
		fmt.Printf("slice=%q\n", slice.Value())
		return nil
	}

	c.Run(os.Args)
}
```

When using `v2.5.0` we get:
```
$ go run ./main.go --slice-flag=foo,bar,baz
slice=["foo,bar,baz"]
$ go run ./main.go --slice-flag=foo,bar,baz --slice-flag=boom
slice=["foo,bar,baz" "boom"]
```

When updating to `v2.5.1` we get:
```
$ go run ./main.go --slice-flag=foo,bar,baz
slice=["foo" "bar" "baz"]
$ go run ./main.go --slice-flag=foo,bar,baz --slice-flag=boom
slice=["foo" "bar" "baz" "boom"]
```
clearly indicating that a bug was fixed for handling comma-separated repeated arguments when updating to at least `v2.5.1`.

We were however depending on the broken behaviour, meaning that when a user ran:
```
nvidia-ctk config --set nvidia-container-runtime.runtimes=crun,runc
```
we had a slice with the following contents:
```
[]string{"nvidia-container-runtime.runtimes=crun,runc"}
```
When we could process -- first splitting by `=` and then the second argument by `,` to generate the eventually slice `[]string{"crun", "runc"}`.

With the update to `v2.27.1` when a user runs:
```
nvidia-ctk config --set nvidia-container-runtime.runtimes=crun,runc
```
we have a slice with the following contents:
```
[]string{"nvidia-container-runtime.runtimes=crun" ,"runc"}
```
Which we cannot effectively parse.

With the switch to `;` as our own separator, we no have the inputs:
```
nvidia-ctk config --set nvidia-container-runtime.runtimes=crun;runc
```
which would be parsed to:
```
[]string{"nvidia-container-runtime.runtimes=crun;runc"}
```
meaning that we can do the same processing as before.

Reply:
OK, so this is defining the separator we want to use to break apart elements of an *embedded* list of strings that we don't want to become elements of the overall `StringSlice`.

Reply:
Meaning that `,` is still the default for the top-level separator, and we now defining `;` as the embedded separator.

Reply:
I think this is fine then -- my only worry would be if / where else we rely on this "bug". I seem to remember relying on this functionality in the device-plugin as well, for example.

Reply:
Seems reasonable to me